src/main/scala/com/gu/fulfilmentLookup/SalesforceCaseService.scala

```diff
@@ -0,0 +1,100 @@
+
+import java.util.concurrent.TimeUnit
+import okhttp3._
+import play.api.libs.json.{ JsPath, JsSuccess, Json, Reads }
+import play.api.libs.functional.syntax._
+import scalaz.{ -\/, \/, \/- }
+
+case class SalesforceAuth(accessToken: String, instanceUrl: String)
+
+object SalesforceAuth {
+
+  implicit val salesforceAuthReads: Reads[SalesforceAuth] = (
+    (JsPath \ "access_token").read[String] and
+    (JsPath \ "instance_url").read[String]
+  )(SalesforceAuth.apply _)
+
+}
+
+trait CaseService {
+  def authenticate(config: Config): String \/ SalesforceAuth
+  def raiseCase(config: Config, lookupRequest: LookupRequest, lookupResponseBody: LookupResponseBody): String \/ Boolean
+}
+
+object SalesforceCaseService extends CaseService with Logging {
+
+  val restClient = new OkHttpClient().newBuilder()
+    .readTimeout(15, TimeUnit.SECONDS)
+    .build()
+
+  def requestBuilder(config: Config, route: String): Request.Builder = {
+    new Request.Builder()
+      .url(s"${config.salesforceUrl}/$route")
+  }
+
+  def withSfAuth(requestBuilder: Request.Builder, salesforceAuth: SalesforceAuth): Request.Builder = {
+    requestBuilder.addHeader("Authorization", s"Bearer ${salesforceAuth.accessToken}")
+  }
+
+  def description(lookupResponseBody: LookupResponseBody): String = {
+
+    val addressInformation = lookupResponseBody.addressDetails.map {
+      address => s"We asked our fulfilment partner to send the paper to: $address"
+    }.getOrElse("")
+
+    "This case has been automatically raised due to a customer-initiated distribution check \n" +
+      "\n" +
+      s"Fulfilment File checked: ${lookupResponseBody.fileChecked} \n" +
+      s"Subscription Number included in file: ${lookupResponseBody.subscriptionInFile} \n" +
+      addressInformation
+
+  }
+
+  override def authenticate(config: Config): String \/ SalesforceAuth = {
+    val builder = requestBuilder(config, "/services/oauth2/token")
+    val formBody = new FormBody.Builder()
+      .add("client_id", config.salesforceClientId)
+      .add("client_secret", config.salesforceClientSecret)
+      .add("username", config.salesforceUsername)
+      .add("password", config.salesforcePassword + config.salesforceToken)
+      .add("grant_type", "password")
+      .build()
+    val request = builder.post(formBody).build()
+    logger.info(s"Attempting to perform Salesforce Authentication")
+    val response = restClient.newCall(request).execute()
+    val responseBody = Json.parse(response.body().string())
+    responseBody.validate[SalesforceAuth] match {
+      case JsSuccess(result, _) =>
+        logger.info(s"Successful Salesforce authentication.")
+        \/-(result)
+      case _ =>
+        -\/(s"Failed to authenticate with Salesforce | body was: ${responseBody.toString}")
+    }
+  }
+
+  override def raiseCase(config: Config, lookupRequest: LookupRequest, lookupResponseBody: LookupResponseBody): String \/ Boolean = {
+    val salesforceAuth = authenticate(config)
```

Comment:
Currently we re-authenticate every time we raise a case, which is not very efficient (it's an extra network call), but I wanted to get something up and running before worrying too much about the details...

Reply:
seems fine to me.  we could always worry later about caching in s3 or dynamo.

Reply:
I'm probably going to steal this code for the job to produce stats about cases raised (assuming you don't beat me to it...) 😛

Reply:
I borrowed most of it from: https://github.com/guardian/membership-common/blob/master/src/main/scala/com/gu/salesforce/Scalaforce.scala#L184 

(I don't want to import membership-common, or deal with the Akka stuff here)

Reply:
I suppose I could also run authenticate inside a future as soon as the lambda starts up and then pass a SalesforceAuth into raiseCase if performance turns out to be a problem...